x/ibc/02-client/keeper/client.go

```diff
@@ -50,15 +52,27 @@ func (k Keeper) CreateClient(
 }
 
 // UpdateClient updates the consensus state and the state root from a provided header
-func (k Keeper) UpdateClient(ctx sdk.Context, clientID string, header exported.Header) error {
+func (k Keeper) UpdateClient(
+	ctx sdk.Context, clientID string, oldHeader, newHeader exported.Header,
+) error {
 	clientType, found := k.GetClientType(ctx, clientID)
 	if !found {
 		return sdkerrors.Wrapf(types.ErrClientTypeNotFound, "cannot update client with ID %s", clientID)
 	}
 
-	// check that the header consensus matches the client one
-	if header.ClientType() != clientType {
-		return sdkerrors.Wrapf(types.ErrInvalidConsensus, "cannot update client with ID %s", clientID)
+	// check that the headers client type matches the stored one
+	if oldHeader.ClientType() != clientType {
+		return sdkerrors.Wrapf(
+			types.ErrInvalidClientType,
+			"old header type ≠ client type (%s ≠ %s)", oldHeader.ClientType().String(), clientType.String(),
+		)
+	}
+
+	if newHeader.ClientType() != clientType {
+		return sdkerrors.Wrapf(
+			types.ErrInvalidClientType,
```

Comment:
Maybe the `ErrInvalidClientType` takes the two types and this string can be housed there?